modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -276,4 +284,22 @@ private void trackSuccessfulRequest(HttpRequest httpRequest, HttpResponse httpRe
             }
         }
     }
+
+    // The underlying http client uses this as part of the connection pool key,
+    // hence we need to use the same instance across all the client instances
+    // to avoid creating multiple connection pools.
+    record NioLoopResources(EventLoopGroup eventLoopGroup) implements LoopResources {
+        @Override
+        public EventLoopGroup onServer(boolean useNative) {
+            // We rely on Nio here, hence we ignore the useNative flag
+            return eventLoopGroup;
+        }
+    }
+
+    static class Resolver extends AddressResolverGroup<InetSocketAddress> {
```

Comment:
Isn't this resolver is the same as `DefaultAddressResolverGroup.INSTANCE`?

Reply:
You're right, this is a leftover from when I was tracking down the issue with the connection pool 👍